src/engine.ts

```diff
@@ -210,6 +210,40 @@ export function initializePolyfill() {
     attributeOldValue: true,
   });
 
+  const originalAttachShadow = Element.prototype.attachShadow;
+  Element.prototype.attachShadow = function (options) {
+    const shadow = originalAttachShadow.apply(this, [options]);
+    mutationObserver.observe(shadow, {
+      childList: true,
+      subtree: true,
+      attributes: true,
+      attributeOldValue: true,
+    });
+    return shadow;
+  }
+
+  const originalReplaceSync = CSSStyleSheet.prototype.replaceSync;
```

Comment:
I think it would be sufficient to just not monkeypatch these if they aren't already defined.